### ISSUE TITLE
Plugin doesnt identify typing extensions

### ISSUE THREAD

**Issue by Goldziher** [snok/flake8-type-checking#115]
the `typing_extensions` lib is identified as a third party package:

```shell
starlite/utils/lifecycle_hooks.py:16:1: TC002 Move third-party import 'typing_extensions.Literal' into a type-checking block
```

**Comment by sondrelg** [comment]
To my knowledge, the `typing_extensions` package is a third-party package, so that seems expected :slightly_smiling_face: Is it not?

**Comment by Goldziher** [comment]
> To my knowledge, the `typing_extensions` package is a third-party package, so that seems expected :slightly_smiling_face: Is it not?

It offers type hints for backwards compatibility

**Comment by sondrelg** [comment]
Third party in this context just means not shipped with Python, and not a part of the local application. That seems right here I think

**Comment by Goldziher** [comment]
> Third party in this context just means not shipped with Python, and not a part of the local application. That seems right here I think

`Literal` accepts a value, `Literal["Moishe"]`, so should I do `Optional["Literal["Moishe"]"]?

**Comment by sondrelg** [comment]
You should be able to do `'Literal["Moishe"]'` while putting the `Literal` import in a type-checking block :+1:
